File: DP/11052.cpp
#define MAX 1001
#include <bits/stdc++.h>

using namespace std;

int main(void) {
	int n, dp[MAX], data[MAX];
	scanf("%d", &n);


	for (int i = 0; i < n; i++) {
		scanf("%d", &data[i]);
	}
	dp[0] = data[0];

	for (int i = 1; i < n; i++) {
		dp[i] = max(data[i], dp[i - 1] + dp[0]);
		for (int j = 0; j < i / 2 + 1; j++) {
			dp[i] = max(dp[i], dp[j] + dp[i - j - 1]);
		}
 	}

	printf("%d", dp[n - 1]);
	return 0;
}

File: DP/11052.py
n = int(input())
data = list(map(int, input().split()))
dp = [0] * n
dp[0] = data[0]

for i in range(1, n):
  dp[i] = max(data[i], dp[i-1] + dp[0])
  for j in range(i//2+1):
    dp[i] = max(dp[i], dp[j] + dp[i-j-1])

print(dp[n-1])
